From 6247764f5e9a9d0e8f8d0c968236fb4d1d66df84 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Fri, 31 Mar 2023 16:50:55 +0100
Subject: [PATCH] fix-win-build

---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5f47d3d..c10f73f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,13 +72,13 @@ find_package(PkgConfig)
 if(PKG_CONFIG_FOUND)
     pkg_check_modules(LIBUSB libusb-1.0 IMPORTED_TARGET)
 else()
-    set(LIBUSB_LIBRARIES "" CACHE STRING "manual libusb path")
-    set(LIBUSB_INCLUDE_DIRS "" CACHE STRING "manual libusb includepath")
+    set(LIBUSB_LIBRARIES "${CMAKE_INSTALL_PREFIX}/lib/libusb-1.0.lib" CACHE STRING "manual libusb path")
+    set(LIBUSB_INCLUDE_DIRS "${CMAKE_INSTALL_PREFIX}/include/libusb-1.0" CACHE STRING "manual libusb includepath")
 endif()
 
 if(MSVC)
-    set(THREADS_PTHREADS_LIBRARY "" CACHE STRING "manual pthread-win32 path")
-    set(THREADS_PTHREADS_INCLUDE_DIR "" CACHE STRING "manual pthread-win32 includepath")
+    set(THREADS_PTHREADS_LIBRARY "${CMAKE_INSTALL_PREFIX}/lib/pthreadVC3.lib" CACHE STRING "manual pthread-win32 path")
+    set(THREADS_PTHREADS_INCLUDE_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE STRING "manual pthread-win32 includepath")
 else()
     set(THREADS_PTHREADS_LIBRARY "" CACHE INTERNAL "manual pthread-win32 path")
     set(THREADS_PTHREADS_INCLUDE_DIR "" CACHE INTERNAL "manual pthread-win32 includepath")
-- 
2.34.1.windows.1

